client/app/domain/zone/delete-all/domain-zone-delete-all.controller.js

```diff
@@ -1,26 +1,44 @@
 angular.module("App").controller(
     "DomainZoneDeleteAllCtrl",
     class DomainZoneDeleteAllCtrl {
-        constructor ($scope, Alerter, Domain) {
+        constructor ($scope, Alerter, Domain, Hosting) {
             this.$scope = $scope;
-            this.Alerter = Alerter;
-            this.Domain = Domain;
+            this.alerter = Alerter;
+            this.domainService = Domain;
+            this.hostingService = Hosting;
         }
 
         $onInit () {
             this.domain = this.$scope.currentActionData;
-            this.loading = false;
+            this.canDeleteAllZone = false;
+            this.loading = true;
             this.$scope.deleteAllZone = () => this.deleteAllZone();
+
+            this.checkAllZoneCanBeDelete(this.domain.name);
+        }
+
+        checkAllZoneCanBeDelete (name) {
+            this.loading = true;
+            return this.hostingService
+                .getHosting(name, null)
+                .then(() => {
+                    this.canDeleteAllZone = false;
+                })
+                .catch((err) => {
+                    this.canDeleteAllZone = err.status === 404;
```

Comment:
I highly doubt you get a `status` property if the service uses `OvhHttp`

Reply:
The service don't